test/stdlib/Unmanaged.swift

```diff
@@ -71,4 +71,22 @@ UnmanagedTests.test("Opaque") {
   _fixLifetime(ref)
 }
 
+UnmanagedTests.test("Opaque avoid retain/release") {
+  // The purpose of this test is to ensure that the fromOpaque/toOpaque calls do
+  // NOT attempt to retain/release the class instance at the passed pointer.
+  // Here we're simulating a dangling pointer referencing a class who has
+  // already been released (the allocated pointer points at nothing) and
+  // attempting to create an Unmanaged instance from it and get back the
+  // pointer. This test's expectEqual is kind of bogus, we're just checking that
+  // it doesn't crash.
+
+  let ref = UnsafeMutablePointer<Int>(capacity: 1)
+  let unmanaged = Unmanaged.fromOpaque(UnsafeRawPointer(ref))
+  let ref2 = unmanaged.toOpaque()
+
+  expectEqual(ref, ref2)
+
+  ref.deallocate()
```

Comment:
`UnsafeMutablePointer<Int>(capacity: 1)` should be a call to the `allocate` factory, not a direct initializer call.

But I think it might be better to directly initialize a dangling pointer:

```suggestion
  // Create a dangling pointer, usually to unmapped memory.
  let ref = UnsafeRawPointer(bitPattern: 1)!
  // Turn it into a dangling unmanaged reference.
  // We expect this not to crash, as this operation isn't 
  // supposed to dereference the pointer in any way.
  let unmanaged = Unmanaged.fromOpaque(ref)
  // Similarly, converting the unmanaged reference back to a 
  // pointer should not ever try to dereference it either.
  let ref2 = unmanaged.toOpaque()
  // ...and we must get back the same pointer.
  expectEqual(ref, ref2)
```